src/components/PositionMarker.js

```diff
@@ -0,0 +1,28 @@
+import { Marker, useMapEvents } from "react-leaflet";
+
+
+const PositionMarker = ({ location, setLocation, markerLocation, setMarkerLocation }) => {
+  useMapEvents({
+    click(e) {
+      if (setLocation && setMarkerLocation) {
+        setMarkerLocation(e.latlng);
+        setLocation(`${e.latlng.lat},${e.latlng.lng}`);
+      }
+    }
+  });
+
+  if (location && typeof location === "string") {
+    const splitPosition = location.split(",");
+    const lat = splitPosition[0];
+    const lon = splitPosition[1];
+    location = [lat, lon];
```

Comment:
```js
const splitPosition = location.split(",");
location = [splitPosition[0], splitPosition[1]];
```
```js
location = location.split(",");
```

But why are you not using setLocation? You dont want to mutate data like this.

Reply:
I think this issue will also be resolved once I have changed the values for position in the API to use two values and not one string